test/controllers/assessments_controller_test.rb

```diff
@@ -0,0 +1,301 @@
+
+  def teardown; end
+
+  def symptoms_param(model_symptoms)
+    model_symptoms.map do |s|
+      {
+        name: s.name,
+        value: s.value,
+        type: s.type,
+        label: s.label,
+        notes: s.notes,
+        required: s.required
+      }
+    end
+  end
+
+  test 'successful get new report as patient' do
+    ADMIN_OPTIONS['report_mode'] = true
+    patient_submission_token = patients(:patient_1).submission_token
+    unique_identifier = patients(:patient_1).jurisdiction.unique_identifier
+    get :new, params: {
+      patient_submission_token: patient_submission_token,
+      unique_identifier: unique_identifier
+    }
+    assert_response :success
+  end
+
+  test 'successful get using old_unique_identifier as patient' do
+    ADMIN_OPTIONS['report_mode'] = true
+    patient_submission_token = patients(:patient_1).submission_token
+    unique_identifier = JurisdictionLookup.find_by(
+      new_unique_identifier: patients(:patient_1).jurisdiction.unique_identifier
+    ).old_unique_identifier
+    get :new, params: {
+      patient_submission_token: patient_submission_token,
+      unique_identifier: unique_identifier
+    }
+    assert_response :success
+  end
+
+  test 'successful get new report as user' do
+    ADMIN_OPTIONS['report_mode'] = false
+    user = create(:public_health_enroller_user)
+    sign_in user
+    patient_submission_token = patients(:patient_1).submission_token
+    unique_identifier = patients(:patient_1).jurisdiction.unique_identifier
+    get :new, params: {
+      patient_submission_token: patient_submission_token,
+      unique_identifier: unique_identifier
+    }
+    assert_response :success
+  end
+
+  test 'successful create report as patient' do
+    ADMIN_OPTIONS['report_mode'] = true
+    patient_submission_token = patients(:patient_1).submission_token
+    unique_identifier = patients(:patient_1).jurisdiction.unique_identifier
+    assert_difference ['AssessmentReceipt.count'], 1 do
+      post :create, params: {
+        patient_submission_token: patient_submission_token,
+        unique_identifier: unique_identifier,
+        experiencing_symptoms: 'yes'
+      }
+    end
+    assert_response :success
+  end
+
+  test 'hit limit number of reports per time period as patient' do
+    ADMIN_OPTIONS['report_mode'] = true
+    patient_submission_token = patients(:patient_1).submission_token
+    unique_identifier = patients(:patient_1).jurisdiction.unique_identifier
+    assert_difference ['AssessmentReceipt.count'], 1 do
+      post :create, params: {
+        patient_submission_token: patient_submission_token,
+        unique_identifier: unique_identifier,
+        experiencing_symptoms: 'yes'
+      }
+    end
+    assert_response :success
+    get :new, params: {
+      patient_submission_token: patient_submission_token,
+      unique_identifier: unique_identifier
+    }
+    assert_redirected_to :already_reported_report
+  end
+
+  test 'successful create report as user' do
```

Comment:
move the loop for different roles outside the test such that the test name gets changed with each role so if a role fails we know which.

Reply:
This should be resolved now